addons/mail/models/discuss/discuss_channel.py

```diff
@@ -359,7 +369,10 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                 user = member.partner_id.user_ids[0] if member.partner_id.user_ids else self.env['res.users']
                 if user:
                     notifications.append((member.partner_id, 'discuss.channel/joined', {
-                        'channel': member.channel_id.with_user(user).with_context(allowed_company_ids=user.company_ids.ids)._channel_info()[0],
+                        'channel': {
+                            **member.channel_id._channel_basic_info(),
```

Comment:
This change makes the other members no longer known.

In `group`, the members need to be known for the name to be correct (when the name field is empty).
This makes it impossible until the member is fetched from another way.

It's ok in the context of this PR, but it needs to be handled separately.
I think it's not a great idea that the name is computed in JS and depends on member.
We should make it so that the "computed" name is fetched from the python always. And updated from bus whenever members change.